### DISCUSSION TITLE
Renew root permission auto close swlan0

### DISCUSSION THREAD

**Discussion by FisherWooD** [Mygod/VPNHotspot#625] [Q&A] [answered]
When i use my phone Some times i get notification magisk root permission vpnhotspot. When i show this notification. My vpn interface (swlan0) auto close. Must need open manually. I check settings but find any option for direct open swlan0 interface auto, how can i fix this problem

**Comment by Mygod** [comment]
Not sure what you quite mean. swlan0 is probably not a VPN interface?

**Comment by FisherWooD** [comment]
![Screenshot_20241115_090328_VPN Hotspot](https://github.com/user-attachments/assets/6a1cc1df-44ac-4d68-b560-f479a120a550)
Some time i use my phone. Magisk notification root permission for vpnhotspot app. When I see this notification, my network clients network down, then i check vpnhotspot app I see slwan0 is disabled.

Sorry my monkey english :)

**Reply by Mygod** [reply]
Make sure you grant root permission to this app. Feel free to write in your native language if that helps.

**Reply by Mygod** [reply]
If you have already granted root, go to "Settings" and export debug information and send it to me.

**Comment by FisherWooD** [comment]
Remove this file for security
33 mb so must zipped. I get this debug file yesterday, when i show same problem a littbit later

**Reply by Mygod** [reply]
Can you try pressing "clean/reapply rules" in settings and see if that fixes anything? (Also what's the issue that you're facing?) Export a new debug information after that if you are still facing issues.

**Reply by FisherWooD** [reply]
> Can you try pressing "clean/reapply rules" in settings and see if that fixes anything? (Also what's the issue that you're facing?) Export a new debug information after that if you are still facing issues.

OK Now clean and wait I show same problem. when I show I will send you file. thanks for interesting

**Reply by Mygod** [reply]
So is your issue that the service was shut down? Try turning off battery optimizations for this app and see if that helps.

**Reply by FisherWooD** [reply]
> So is your issue that the service was shut down? Try turning off battery optimizations for this app and see if that helps.

My problem is this. You can see my screenshot my old reply. 
Some time suddenly (generally when open vpnhotspot after 1-15 minute) magisk notification popup for vpnhotspot app 
then my clients internets drop vpn, when Check app Hotspot work but swlan0(you can see my screenshot) disabled. Must need enable manuel. After work good for hours.

now I clean clean/reapply rules.
And I dont try "turning off battery optimizations" Now disable it for this app. I will wait. if still facing issues, I will send new debug file, İf problem fixed I will mark as answer for battery optimization.

**Comment by Mygod** [comment, marked as answer]
I found this in your logs:
```
11-14 10:38:41.452  1747  2862 I ChimeraAggressivePolicyHandler: Killed on trigger2 : be.mygod.vpnhotspot[5334], freed: 39404, 10273[5334]:G2(0x20001) score=39,8(48 10/10 39404), pids: 5334 / stats: 4 / adjs: 200 
11-14 10:38:41.452  1488  1488 I SDM     : ResourceImpl::SetMaxBandwidthMode: new bandwidth mode=0
11-14 10:38:41.453 29754 29784 E CamX    : [ERROR][HAL    ] camxhal3.cpp:1319 close() after process camera close
11-14 10:38:41.453 29754 29784 E CamX    : [ERROR][HAL    ] camxhal3.cpp:1331 close() Close: overrideCameraClose is 17 , overrideCameraOpen is 0 
11-14 10:38:41.453  1747  2862 I ActivityManager: Killing 5334:be.mygod.vpnhotspot/u0a273 (adj 200): Chimera #2
```

Looks like some Samsung specific mechanics (called `ChimeraAggressivePolicyHandler`) decided to kill the app. I would suspect that turning off battery saving features (especially those only present on Samsung devices) should help.

**Reply by FisherWooD** [reply]
I test 8 hours and multiple times stop and restart all functions and I set flight mode on off. Amazing but not seen problem. thanks for helping all problem battery saving features. When I disable it, all problem fixed
edit: my phone samsung
